### PR TITLE
CI: bump to XCode 16

### PR DESCRIPTION
**Pull request by kmilos**

### CODE REVIEW

**Comment by zisoft** [comment]
Please don't raise the deployment target. We want to support the oldest OS version with the newest Xcode version.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -281,7 +281,7 @@ jobs:
       matrix:
         build:
           - { os: macos-13, xcode: 15.2, deployment: 13.5 } # LLVM16, x86_64
-          - { os: macos-14, xcode: 15.4, deployment: 14.0 } # LLVM16, arm64
```
Comment (zisoft) [inline .github/workflows/nightly.yml:284]: deployment should be left at 14.0

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -385,7 +385,7 @@ jobs:
 
             The Windows package requires Windows with UCRT (Universal C Runtime), which is shipped with Windows 10+. Darktable should also work on Windows 8.1 [on condition that you install this runtime yourself](https://support.microsoft.com/en-us/topic/update-for-universal-c-runtime-in-windows-c0514201-7fe6-95a3-b0a5-287930f3560c).
 
-            The macOS `*-x86_64.dmg` package requires at least macOS 13.5 (Ventura), the `*-arm64.dmg` package requires at least macOS 14.0 (Sonoma).
+            The macOS `*-x86_64.dmg` package requires at least macOS 13.5 (Ventura), the `*-arm64.dmg` package requires at least macOS 14.5 (Sonoma).
```
Comment (zisoft) [inline .github/workflows/nightly.yml:388]: likewise

**Review by zisoft — CHANGES_REQUESTED** [review]

**Comment by zisoft** [comment]
oh, sorry, my fault.
Xcode 16 requires macOS 14.5, so all good!

**Review by zisoft — APPROVED** [review]

**Comment by MStraeten** [comment]
be careful with xcode16 - highway as a dependency for some darktable dependencies won’t build due to a clang issue …
see https://github.com/google/highway/issues/2317

**Comment by zisoft** [comment]
For now we have no pressure to bump the Xcode version.
We will drop support for macOS 13 with the winter release 2025 when macOS 16 will be released.

**Comment by zisoft** [comment]
Thanks!
So for the nightly builds and my release builds we will stay on the macos-14 runner with Xcode 15 to ensure the deployment target of macOS 14.

For the CI we can use the next macos-15 runner with Xcode 16.

**Pull request closed without merging**